Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -456,6 +457,140 @@ ControlPartType RenderTheme::autoAppearanceForElement(RenderStyle& style, const
     return ControlPartType::NoControl;
 }
 
+RefPtr<ControlPart> RenderTheme::createControlPartForRenderer(const RenderObject& renderer) const
+{
+    ControlPartType type = renderer.style().effectiveAppearance();
+
+    switch (type) {
+    case ControlPartType::NoControl:
+    case ControlPartType::Auto:
+        break;
+
+    case ControlPartType::Checkbox:
+    case ControlPartType::Radio:
+    case ControlPartType::PushButton:
+    case ControlPartType::SquareButton:
+    case ControlPartType::Button:
+    case ControlPartType::DefaultButton:
+    case ControlPartType::Listbox:
+    case ControlPartType::Menulist:
+    case ControlPartType::MenulistButton:
+        break;
+
+    case ControlPartType::Meter: {
```

Comment:
We may want to split this up into separate methods like the old `paint` – I worry this method will become very large.

Reply:
Done. A separate function is added to create the MeterPart. I can also use EnumeratedArray if it makes things clearer.